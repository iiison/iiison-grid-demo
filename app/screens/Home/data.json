[
  {
    "size" : "2_2",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg",
    "content" : { 
      "text" : "Play\nWin\nExcite",
      "size" : "xl"
    },
    "button" : {
      "text" : "Get Started!"
    }
  },
  {
    "size" : "1_1",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg"
  },
  {
    "size" : "1_1",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg"
  },
  {
    "size" : "1_1",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg"
  },
  {
    "size" : "1_1",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg"
  },
  {
    "size" : "1_2",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg",
    "content" : {
      "text" : "Live Games\nWith Friends",
      "size" : "l",
      "shouldCenter" : true
    }
  },
  {
    "size" : "1_1",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg"
  },
  {
    "size" : "1_1",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg"
  },
  {
    "size" : "1_1",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg"
  },
  {
    "size" : "1_1",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg",
    "content" : {
      "text" : "Esports",
      "size" : ""
    }
  },
  {
    "size" : "1_2",
    "imgUrl" : "https://i.imgur.com/yJAsBlY.jpg",
    "content" : {
      "text" : "Love Playing?\nJoin the League",
      "size" : ""
    },
    "link" : {
      "url" : "#",
      "text" : "Apply Now!"
    }
  }
]
