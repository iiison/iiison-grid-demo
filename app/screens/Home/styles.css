@import "../../styles/base/_variables";

.homeGrid {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
	grid-auto-rows: minmax(200px, auto);
	grid-auto-flow: dense;
  grid-gap: 1rem;
	padding: 50px;

  .span_1_1 {
    grid-row-end: span 1;
    grid-column-end: span 1;
  }

  .span_2_2 {
    grid-row-end: span 2;
    grid-column-end: span 2;
  }

  .span_1_2 {
    grid-row-end: span 1;
    grid-column-end: span 2;
  }
}

.gridItem {
  background-color: blue;
  position: relative;

  .heroImg {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 100%;
    height: auto;
  }
}

.headingXL {
  font-size: 3.5em;
  margin-bottom: 10px;
}

.headingL {
  font-size: 3em;
}

.heading {
  font-size: 2em;
}

.applyLink {
  color: #fff;
  flex-basis: 100%;
  text-align: right;
  padding-right: 20px;
  margin-top: 30px;
}

