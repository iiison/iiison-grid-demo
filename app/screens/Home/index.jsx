import React from 'react'
import { PropTypes } from 'prop-types'

import { Button } from '$COMPONENTS'
import styles from './styles.css'

const GridItem = ({ children, imageUrl, size }) => {
  const sizeClass = styles[`span_${size}`]

  return (
    <div className={`${styles.gridItem} ${sizeClass} grid-middle`}>
      <img className={styles.heroImg} src={imageUrl} />
      <div className={styles.content}>
        {children}
      </div>
    </div>
  )
}

GridItem.propTypes = {
  children : PropTypes.node,
  imageUrl : PropTypes.string,
  size     : PropTypes.oneOf(['1_1', '1_2', '2_1', '2_2'])
}

const Home = () => {
  return (
    <div className={`grid ${styles.homeGrid}`}>
      <GridItem size='2_2' imageUrl='https://i.imgur.com/yJAsBlY.jpg'>
        <div className={`grid-middle grid-center full-height t-upper`}>
          <div>
            <p className={`${styles.headingXL}`}>Play<br />Win<br /> Excite</p>
            <Button type='primary'>Get Started!</Button>
          </div>
        </div>
      </GridItem>
      <GridItem size='1_1' imageUrl='https://i.imgur.com/yJAsBlY.jpg' />
      <GridItem size='1_1' imageUrl='https://i.imgur.com/yJAsBlY.jpg' />
      <GridItem size='1_1' imageUrl='https://i.imgur.com/yJAsBlY.jpg' />
      <GridItem size='1_1' imageUrl='https://i.imgur.com/yJAsBlY.jpg' />
      <GridItem size='1_2' imageUrl='https://i.imgur.com/yJAsBlY.jpg'>
        <div className={`grid-middle grid-center full-height t-upper`}>
          <p className={`${styles.headingL}`}>Live Games<br />With Friends</p>
        </div>
      </GridItem>
      <GridItem size='1_1' imageUrl='https://i.imgur.com/yJAsBlY.jpg' />
      <GridItem size='1_1' imageUrl='https://i.imgur.com/yJAsBlY.jpg' />
      <GridItem size='1_1' imageUrl='https://i.imgur.com/yJAsBlY.jpg' />
      <GridItem size='1_1' imageUrl='https://i.imgur.com/yJAsBlY.jpg'>
        <div className={`grid-middle grid-center full-height t-upper`}>
          <p className={`${styles.headingL}`}>Esports</p>
        </div>
      </GridItem>
      <GridItem size='1_2' imageUrl='https://i.imgur.com/yJAsBlY.jpg'>
        <div className={`grid-middle grid-center full-height t-upper`}>
          <div className='col grid-center'>
            <p className={`${styles.heading}`}>Love Playing?<br />Join the League</p>
            <a href='#' className={`t-right t-capitalize ${styles.applyLink}`}>Apply Now!</a>
          </div>
        </div>
      </GridItem>
    </div>
  )
}

export default Home

