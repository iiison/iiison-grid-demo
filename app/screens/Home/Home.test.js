import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Home, { GridItem } from './'
import data from './data.json'

describe('>> SCREEN -- Tests Home Page', () => {
  it('Home Page Snapshot Test', () => {
    const component = renderer
      .create(<Home />)
      .toJSON()

    expect(component).toMatchSnapshot();
  })

  it('Counts Grid Items At Home Page', () => {
    // const input = shallow(<PinInputUnit {...props} />)
    const component = shallow(<Home />)

    const result = component.find('GridItem').length
    const expected = 11

    expect(result).toEqual(expected)
  })
})

describe('>> COMPONENT -- Tests home page GridItem', () => {
  it ('GridItem 1_1 Snapshot Test With Content And Button', () => {
    const props = {
      ...data[0],
      size : '1_1'
    }
    const component = renderer
      .create(<GridItem {...props} />)
      .toJSON()

    expect(component).toMatchSnapshot();
  })

  it ('GridItem 1_1 Snapshot Test Without Content and Button', () => {
    const props = {
      ...data[0],
      content : undefined,
      button  : undefined
    }
    const component = renderer
      .create(<GridItem {...props} />)
      .toJSON()

    expect(component).toMatchSnapshot();
  })

  it ('GridItem 1_1 Snapshot Test With Content and Link', () => {
    const props = {
      ...data[10],
      size : '1_1'
    }
    const component = renderer
      .create(<GridItem {...props} />)
      .toJSON()

    expect(component).toMatchSnapshot();
  })

  it ('GridItem 1_2 Snapshot Test', () => {
    const props = {
      ...data[0],
      size : '1_2'
    }
    const component = renderer
      .create(<GridItem {...props} />)
      .toJSON()

    expect(component).toMatchSnapshot();
  })

  it ('GridItem 2_2 Snapshot Test', () => {
    const props = { ...data[0] }
    const component = renderer
      .create(<GridItem {...props} />)
      .toJSON()

    expect(component).toMatchSnapshot();
  })
})

