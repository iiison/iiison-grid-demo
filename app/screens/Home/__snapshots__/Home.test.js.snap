// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> COMPONENT -- Tests home page GridItem GridItem 1_1 Snapshot Test With Content And Button 1`] = `
<div
  className="gridItem span_1_1 grid-middle"
>
  <img
    className="heroImg"
    src="https://i.imgur.com/yJAsBlY.jpg"
  />
  <div
    className="content"
  >
    <div
      className="grid-middle grid-center full-height t-upper"
    >
      <div>
        <p
          className="headingXL undefined"
        >
          Play
        </p>
        <p
          className="headingXL undefined"
        >
          Win
        </p>
        <p
          className="headingXL undefined"
        >
          Excite
        </p>
        <button
          className="primary button "
        >
          Get Started!
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`>> COMPONENT -- Tests home page GridItem GridItem 1_1 Snapshot Test With Content and Link 1`] = `
<div
  className="gridItem span_1_1 grid-middle"
>
  <img
    className="heroImg"
    src="https://i.imgur.com/yJAsBlY.jpg"
  />
  <div
    className="content"
  >
    <div
      className="grid-middle grid-center full-height t-upper"
    >
      <div>
        <p
          className="heading undefined"
        >
          Love Playing?
        </p>
        <p
          className="heading undefined"
        >
          Join the League
        </p>
      </div>
      <a
        className="t-right t-capitalize col-12 applyLink"
        href="#"
      >
        Apply Now!
      </a>
    </div>
  </div>
</div>
`;

exports[`>> COMPONENT -- Tests home page GridItem GridItem 1_1 Snapshot Test Without Content and Button 1`] = `
<div
  className="gridItem span_2_2 grid-middle"
>
  <img
    className="heroImg"
    src="https://i.imgur.com/yJAsBlY.jpg"
  />
  <div
    className="content"
  />
</div>
`;

exports[`>> COMPONENT -- Tests home page GridItem GridItem 1_2 Snapshot Test 1`] = `
<div
  className="gridItem span_1_2 grid-middle"
>
  <img
    className="heroImg"
    src="https://i.imgur.com/yJAsBlY.jpg"
  />
  <div
    className="content"
  >
    <div
      className="grid-middle grid-center full-height t-upper"
    >
      <div>
        <p
          className="headingXL undefined"
        >
          Play
        </p>
        <p
          className="headingXL undefined"
        >
          Win
        </p>
        <p
          className="headingXL undefined"
        >
          Excite
        </p>
        <button
          className="primary button "
        >
          Get Started!
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`>> COMPONENT -- Tests home page GridItem GridItem 2_2 Snapshot Test 1`] = `
<div
  className="gridItem span_2_2 grid-middle"
>
  <img
    className="heroImg"
    src="https://i.imgur.com/yJAsBlY.jpg"
  />
  <div
    className="content"
  >
    <div
      className="grid-middle grid-center full-height t-upper"
    >
      <div>
        <p
          className="headingXL undefined"
        >
          Play
        </p>
        <p
          className="headingXL undefined"
        >
          Win
        </p>
        <p
          className="headingXL undefined"
        >
          Excite
        </p>
        <button
          className="primary button "
        >
          Get Started!
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`>> SCREEN -- Tests Home Page Home Page Snapshot Test 1`] = `
<div
  className="grid homeGrid"
>
  <div
    className="gridItem span_2_2 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    >
      <div
        className="grid-middle grid-center full-height t-upper"
      >
        <div>
          <p
            className="headingXL undefined"
          >
            Play
          </p>
          <p
            className="headingXL undefined"
          >
            Win
          </p>
          <p
            className="headingXL undefined"
          >
            Excite
          </p>
          <button
            className="primary button "
          >
            Get Started!
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="gridItem span_1_1 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    />
  </div>
  <div
    className="gridItem span_1_1 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    />
  </div>
  <div
    className="gridItem span_1_1 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    />
  </div>
  <div
    className="gridItem span_1_1 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    />
  </div>
  <div
    className="gridItem span_1_2 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    >
      <div
        className="grid-middle grid-center full-height t-upper"
      >
        <div>
          <p
            className="headingL t-center"
          >
            Live Games
          </p>
          <p
            className="headingL t-center"
          >
            With Friends
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="gridItem span_1_1 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    />
  </div>
  <div
    className="gridItem span_1_1 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    />
  </div>
  <div
    className="gridItem span_1_1 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    />
  </div>
  <div
    className="gridItem span_1_1 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    >
      <div
        className="grid-middle grid-center full-height t-upper"
      >
        <div>
          <p
            className="heading undefined"
          >
            Esports
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="gridItem span_1_2 grid-middle"
  >
    <img
      className="heroImg"
      src="https://i.imgur.com/yJAsBlY.jpg"
    />
    <div
      className="content"
    >
      <div
        className="grid-middle grid-center full-height t-upper"
      >
        <div>
          <p
            className="heading undefined"
          >
            Love Playing?
          </p>
          <p
            className="heading undefined"
          >
            Join the League
          </p>
        </div>
        <a
          className="t-right t-capitalize col-12 applyLink"
          href="#"
        >
          Apply Now!
        </a>
      </div>
    </div>
  </div>
</div>
`;
